### DISCUSSION TITLE
Refiners/Filters do not get reset when a new Search is performed using the Search Box(unlike v3)

### DISCUSSION THREAD

**Discussion by kasgit** [microsoft-search/pnp-modern-search#3031] [Ideas]
**Version used**
4.5.3

**Describe the bug**
We upgraded the Search experience from v3 to v4. Since then, users have started complaining about the new refiner experience. In V3, following scenarios would reset the refiner selections.

1. Changing verticals
2. When a new keyword is entered in the Search Box

However, this is not the behaviour in v4.

**To Reproduce**
1. Make sure a Search Box webpart, a Refiner webpart & a Results webpart are present on the page and connected.
2. Perform some search and select a refiner to narrow down the results.
3. Search for another keyword using the Search Box.
4. The refiners will not reset which affect the displayed result set.

**Expected behavior**
The selected refiners should be cleared according to the new result set. 

**Desktop (please complete the following information):**
 - Chrome

**Additional context**
Users see this as a basic requirement which used to be there, but has been removed since v4. It particularly causes confusion when the verticals are connected to same filters but different result sets are provided in each vertical. Sometimes, there will be no result for a new keyword search because some refiner is selected. PnP search webparts are great replacement for the plain OOTB search experience, but this one suboptimal experience is causing a lot of confusions for users to the extent the users need to be trained to use Search.

**Comment by FranckyC** [comment]
This is by design following the OOTB modern search experience. Maybe we could add a flag in the settings of the search results Web Part.

**Comment by kasgit** [comment]
> 

Hi FranckyC, thanks for your response. People would use this webpart mostly because they're not happy with OOTB experience. So, not sure why it has been designed based on the OOTB experience. 

When the page has 4 or 5 verticals without any relationship between them but only metadata, the webpart provides a very bad user experience for new search keyword with selected filters. We conducted a number of user experience tests and users get baffled when they see no results because of filter selections after they enter a new keyword. Then they have to manually go and clear the filters. Otherwise, it is a great webpart.

**Comment by gary-walker-82** [comment]
@FranckyC in the last few days I’ve noticed a change in this behaviour where the filters are no longer retained when the Microsoft search box is changed. This is behaviour we relayed on, as we had a client who wanted to filter before they searched. 
Are you working on this enhancement to be able to choose to retain the filters?  If not would you except a PR for it?

**Comment by wobba** [comment]
@gary-walker-82 are you saying you have not changed anything, retaining filters was working and now it's not? If so, that seems to be an API issue then?
